src/auth.rs

```diff
@@ -17,6 +21,53 @@ pub fn parse_auth(auth: String, host: &str) -> io::Result<(String, Option<String
     }
 }
 
+pub fn read_netrc() -> Option<String> {
+    let mut netrc_buf = String::new();
+    if let Some(mut hd_path) = home_dir() {
+        hd_path.push(".netrc");
+        if let Ok(mut netrc_file) = File::open(hd_path) {
+            if netrc_file.read_to_string(&mut netrc_buf).is_ok() {
+                return Some(netrc_buf);
+            };
+        };
```

Comment:
This can be done in one go with [`std::fs::read_to_string`](https://doc.rust-lang.org/std/fs/fn.read_to_string.html)